presto-main/src/main/java/io/prestosql/sql/planner/optimizations/PredicatePushDown.java

```diff
@@ -501,21 +518,30 @@ public PlanNode visitJoin(JoinNode node, RewriteContext<Expression> context)
                         rightProjections.put(rightSymbol, rightExpression);
                     }
 
-                    equiJoinClauses.add(new JoinNode.EquiJoinClause(leftSymbol, rightSymbol));
+                    ComparisonExpression.Operator operator = alignedComparison ? comparison.getOperator() : comparison.getOperator().flip();
+                    ComparisonExpression filter = new ComparisonExpression(operator, leftSymbol.toSymbolReference(), rightSymbol.toSymbolReference());
+                    dynamicFilteringClauses.add(filter);
+                    if (isEquality) {
+                        equiJoinClauses.add(new JoinNode.EquiJoinClause(leftSymbol, rightSymbol));
+                    }
+                    else {
+                        joinFilterBuilder.add(filter);
+                    }
                 }
                 else {
+                    // Inequality join predicate && dynamic filtering is disabled.
                     joinFilterBuilder.add(conjunct);
                 }
             }
 
-            DynamicFiltersResult dynamicFiltersResult = createDynamicFilters(node, equiJoinClauses, session, idAllocator);
+            DynamicFiltersResult dynamicFiltersResult = createDynamicFilters(node, dynamicFilteringClauses.build(), session, idAllocator);
```

Comment:
please pass `newJoinFilter` instead

Reply:
I think that `newJoinFilter` would contain only the inequality predicates - but `createDynamicFilters` needs also the equi-join predicates. For example, consider the following query:
```sql
SELECT * FROM L, R WHERE L.x = R.x AND L.y < R.y;
```
In this case, `createDynamicFilters` needs `L.x = R.x AND L.y < R.y` but `newJoinFilter` will contain only `L.y < R.y`, IIUC.

Reply:
Added both `equiJoinClauses ` and `newJoinFilter`.